tools/shell/shell_runner.cpp

```diff
@@ -37,11 +43,23 @@ int main(int argc, char* argv[]) {
     if (readOnlyMode) {
         systemConfig.readOnly = true;
     }
+    if (!pathToHistory.empty() && pathToHistory.back() != '/') {
+        pathToHistory += '/';
+    }
+    pathToHistory += "history.txt";
+    std::unique_ptr<VirtualFileSystem> vfs = std::make_unique<VirtualFileSystem>();
+    try {
+        std::unique_ptr<FileInfo> fp = vfs->openFile(pathToHistory, O_CREAT);
```

Comment:
What if the file already exists?
You may have to use `O_CREAT | O_WRONLY ` here.

Reply:
If the file already exists then the O_CREAT flag has no effect and the file gets opened as normal and the content of the file is preserved.